db/seeds_for_general_election.rb

```diff
@@ -24,6 +24,13 @@
   puts party_attributes.slice(:name, :smv_code)
 end
 
+unique_smv_codes = Party.all.each_with_object(Set.new) { |p, set| set.add(p.smv_code) }
```

Comment:
Nevertheless, it wouldn't cost anything significant to add a db constraint for this too, would it?  I would have thought better to have defence in depth.

Reply:
These tables only ever change at seed time which might be
* initialising database
* re-runs for updated party list, or new recommendations, etc.
So the checks would catch any problems

My gut feeling (and vague recollection) is that adding foreign key constraints to tables accessed via activeRecord  only leads to problems. You can find yourself fixing problems that only exist because of the constraints. (Could be that I'm thinking of rails before activeRecord supported foreign keys though.

On balance I thought these checks were a better way to go than foreign key constraints.

Reply:
> These tables only ever change at seed time which might be
> 
> * initialising database
> * re-runs for updated party list, or new recommendations, etc.
>   So the checks would catch any problems

Sure - to be clear, it's great to have this check and I'm certainly not arguing against having it :)

> My gut feeling (and vague recollection) is that adding foreign key constraints to tables accessed via activeRecord only leads to problems. You can find yourself fixing problems that only exist because of the constraints. (Could be that I'm thinking of rails before activeRecord supported foreign keys though.
> 
> On balance I thought these checks were a better way to go than foreign key constraints.

I can imagine operations like deletion via activeRecord possibly running into problems if it didn't handle foreign keys correctly (and funnily enough see [this ancient answer I wrote](https://stackoverflow.com/a/7805719/179332) which must be hopelessly out of date by now).  However, we're not talking about foreign keys here, just a uniqueness constraint within the table.